File: C/unit1/selection_sort.c
// import header files
#include <stdio.h>

// buiding swap function to swap two values
void swap(int *x, int *y)
{
    int temp = *x;
    *x = *y;
    *y = temp;
}

// selection sort algorithm
void selsort(int a[], int n)
{
    int min;
    for (int i = 0; i < n - 1; i++)
    {
        min = i;
        for (int j = i + 1; j < n; j++)
        {
            if (a[j] < a[min])
            {
                min = j;
            }
        }
        swap(&a[min], &a[i]);
    }
}
// main function to show how it works
int main()
{
    int arr[] = {12, 1, 45, 90, 14, 35, 89, 16, 20, 29};
    int size = sizeof(arr) / sizeof(arr[0]);
    selsort(arr, size);
    for (int i = 0; i < size; i++)
    {
        printf("%d\n", arr[i]);
    }
}

File: Java/unit1/selection_sort.java
package Java.unit1;

public class selection_sort {
    void sort(int arr[]) {
        int size = arr.length;
        for (int i = 0; i < size - 1; i++) {
            int min = i;
            for (int j = i + 1; j < size; j++) {
                if (arr[j] < arr[min]) {
                    min = j;
                }
            }
            int temp = arr[min];
            arr[min] = arr[i];
            arr[i] = temp;
        }
    }

    public static void main(String args[]) {
        selection_sort ob = new selection_sort();
        int arr[] = { 12, 1, 45, 90, 14, 35, 89, 16, 20, 29 };
        ob.sort(arr);
        for (int i = 0; i < arr.length; i++) {
            System.out.println(arr[i]);
        }
    }

}